pkg/resource/deploy/step.go

```diff
@@ -945,7 +984,17 @@ func (s *ImportStep) Diffs() []resource.PropertyKey                { return s.di
 func (s *ImportStep) DetailedDiff() map[string]plugin.PropertyDiff { return s.detailedDiff }
 
 func (s *ImportStep) Apply(preview bool) (resource.Status, StepCompleteFunc, error) {
-	complete := func() { s.reg.Done(&RegisterResult{State: s.new}) }
+	complete := func() {
+		now := time.Now().UTC()
+		s.new.Modified = &now
+		s.new.Created = nil
+
+		if s.old != nil && s.new.ID == s.old.ID {
+			s.new.Created = s.old.Created
+			s.new.Modified = s.old.Modified
```

Comment:
I must be misunderstanding this.
Per 0f30e8198, modifies is:

```
Modified: timestamp referring to when pulumi's state is reconciled with with
the provider (i.e. via an update, refresh, import)
```

So the `new.Modified = &now` makes sense, but I'm not sure I understand why `new.Modified = old.Modified`.
Wouldn't this still represent a state reconciliation if the properties have changed?
Can the properties not change from old state for import?